tests/queries/0_stateless/01676_clickhouse_client_autocomplete.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+function test_completion_word()
+{
+    local w=$1 && shift
+
+    local w_len=${#w}
+    local compword_begin=${w:0:$((w_len-3))}
+    local compword_end=${w:$((w_len-3))}
+
+    # NOTE: here and below you should escape variables of the expect.
+    timeout 10s expect << EOF
+log_user 0
+set timeout 1
+match_max 100000
+# A default timeout action is to do nothing, change it to fail
+expect_after {
+    timeout {
+        exit 1
+    }
+}
+
+spawn bash -c "$CLICKHOUSE_CLIENT_BINARY $CLICKHOUSE_CLIENT_OPT"
+expect ":) "
+
+# Make a query
+send -- "SET $compword_begin"
+expect "SET $compword_begin"
+
+# Wait for suggestions to load, they are loaded in background
+set is_done 0
+while {\$is_done == 0} {
+    send -- "\\t"
+    expect {
+        "$compword_begin$compword_end" {
```

Comment:
Maybe just `$compword_end`

It completes instantly in non-ambiguous case, and displays colored list of variants in other cases.
Your example looks like to fail in both cases (maybe you are using readline that does not have colored suggestions).

Reply:
>Maybe just $compword_end

Initially I wrote like this, but this not work correctly for cases like `foobar_aaaaaaa`, since last 3 bytes is the same.

> It completes instantly in non-ambiguous case, and displays colored list of variants in other cases.

Yes, but firstly it completes the base part, i.e. `forc` -> `force_` w/o coloring and display list of suggestions

> Your example looks like to fail in both cases

That's how it's meant to be (stricter is better), and the variants for completion was selected that way.
I think that the failures was due to too small timeout, increased, let's see.

> (maybe you are using readline that does not have colored suggestions).

Too big difference, I stopped doing this.